Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts

```diff
@@ -22,155 +22,852 @@ var username = serverEndpointAuth['parameters']['username'];
 var password = serverEndpointAuth['parameters']['password'];
 
 var jobName = tl.getInput('jobName', true);
-var jobQueueUrl = serverEndpointUrl + '/job/' + jobName + '/build';
-tl.debug('jobQueueUrl=' + jobQueueUrl);
 
 var captureConsole = tl.getBoolInput('captureConsole', true);
-var captureConsolePollInterval = 5000; // five seconds is what the Jenkins Web UI uses
+var pollInterval = 5000; // five seconds is what the Jenkins Web UI uses
+
+// capturePipeline is only possible if captureConsole mode is enabled
+var capturePipeline = captureConsole ? tl.getBoolInput('capturePipeline', true) : false;
+
+var parameterizedJob = tl.getBoolInput('parameterizedJob', true);
+
+var jobQueueUrl = addUrlSegment(serverEndpointUrl, '/job/' + jobName);
+jobQueueUrl += (parameterizedJob) ? '/buildWithParameters?delay=0sec' : '/build?delay=0sec';
+tl.debug('jobQueueUrl=' + jobQueueUrl);
+
+function addUrlSegment(baseUrl: string, segment: string): string {
+    var resultUrl = null;
+    if (baseUrl.endsWith('/') && segment.startsWith('/')) {
+        resultUrl = baseUrl + segment.slice(1);
+    } else if (baseUrl.endsWith('/') || segment.startsWith('/')) {
+        resultUrl = baseUrl + segment;
+    } else {
+        resultUrl = baseUrl + '/' + segment;
+    }
+    return resultUrl;
+}
+
+function failError(err): void {
+    fail(err);
+}
 
 function failReturnCode(httpResponse, message: string): void {
     var fullMessage = message +
         '\nHttpResponse.statusCode=' + httpResponse.statusCode +
         '\nHttpResponse.statusMessage=' + httpResponse.statusMessage +
         '\nHttpResponse=\n' + JSON.stringify(httpResponse);
-    tl.debug(fullMessage);
-    tl.setResult(tl.TaskResult.Failed, fullMessage);
+    fail(fullMessage);
 }
 
-// These are set once the job is successfully queued and don't change afterwards
-var jenkinsTaskName;
-var jenkinsExecutableNumber
-var jenkinsExecutableUrl;
+function fail(message: string): void {
+    tl.debug('fail');
+    tl.debug(message);
+    tl.setResult(tl.TaskResult.Failed, message);
+    tl.exit(1);
```

Comment:
NO! never exit the process.  That's deprecated.  It's unsafe as it kills the process and can have timing issues losing output inclusing critical status.  Instead use reliable async code that naturally lets the script execute.